File: semana-1/ex001.js
class BTree () {
    function constructor(
        nodes = [];
    ) {}
   // o que deve estar em um constructor e o que deve estar fora dele? 

    // Retorna informações do nó p da árvore
    function getInfo(p) {
        return self.data;
    }
    function getLeft(p) {
        return self.left
    }

    function getRight(p) {
        return self.right
    }

    function getFather(p) {
        return self.father
    }
}

class Node() {
    function constructor() {
    }
}

// Criando uma árvore

const input = [1, 2, 3, 4]

File: semana-1/ex001.py
class Node():
    def __init__(self, data = None, left = None, right = None, father = None):
        self.left = left
        self.right = right
        self.father = father
        self.data = data # este é um campo generico, para guardar informacões satelites)

    # Getters
    def getInfo(self): return self.data
    def getLeft(self): return self.left
    def getRight(self): return self.right
    def getFather(self): return self.father

class BTree():
    def __init__(self) -> None:
        self.root = None

    def maximum(self, node: Node) -> Node:
        if node == None:
            return None
        atual = node
        while atual.getRight() is not None:
            atual = atual.getRight()
        return atual

    def minimum(self, node : Node) -> Node:
        if node == None:
            return None
        atual = node
        while atual.getLeft() is not None:
            atual = atual.getLeft()
        return atual

    def sucessor(self, node: Node) -> Node:
        if node == None:
            return
        node.getRight()
        return

    def predecessor(self, node: Node) -> Node:
        if node == None:
            return
        node.getLeft()
        return

    def insert(self, node: Node) -> None:
        # Se a árvore estiver vazia
        if self.root == None:
            self.root = node

        